toolbox/gpu_operator.py

```diff
@@ -0,0 +1,188 @@
+from toolbox._common import PlaybookRun
+import secrets
+
+
+class GPUOperator:
+    """
+    Commands for deploying, building and testing the GPU operator in various ways
+    """
+    @staticmethod
+    def deploy_from_commit(git_repository, git_reference, tag_uid=None):
+        """
+        Deploys the GPU operator from the given git commit
+
+        Args:
+            git_repository: The git repository to deploy from, e.g. https://github.com/NVIDIA/gpu-operator.git
+            git_reference: The git ref to deploy from, e.g. master
+            tag_uid: The GPU operator image tag UID. See 'oc get imagestreamtags -n gpu-operator-ci -oname' for the tag-uid to reuse
+        """
+
+        if tag_uid is None:
+            tag_uid = secrets.token_hex(4)
+
+        opts = {
+            "gpu_operator_git_repo": git_repository,
+            "gpu_operator_git_ref": git_reference,
+            "gpu_operator_image_tag_uid": tag_uid,
+        }
+
+        return PlaybookRun("gpu_operator_deploy_custom_commit", opts)
+
+    @staticmethod
+    def deploy_cluster_policy():
+        """
+        Create the ClusterPolicy from the CSV
+        """
+        print("Creating the ClusterPolicy from the CSV")
+        return PlaybookRun(
+            "gpu_operator_deploy_from_operatorhub",
+            {"gpu_operator_deploy_from": "pre-deployed"},
+        )
+
+    @staticmethod
+    def deploy_from_bundle(bundle=None):
+        """
+        Deploys the GPU Operator from a bundle
+        """
+        opts = {"gpu_operator_deploy_from": "bundle"}
+
+        if bundle is None or bundle == 'master':
+            print("Deploying the GPU Operator from OperatorHub using the master bundle")
+            return PlaybookRun("gpu_operator_deploy_from_operatorhub", opts)
+
+        opts["deploy_bundle_image"] = bundle
+        return PlaybookRun("gpu_operator_deploy_from_operatorhub", opts)
+
+    @staticmethod
+    def deploy_from_operatorhub(version=None, channel=None):
+        """
+        Deploys the GPU operator from OperatorHub
+
+        Args:
+            version: The version to deploy. If unspecified, deploys the latest version available in OperatorHub. Run the toolbox gpu_operator list_version_from_operator_hub subcommand to see the available versions.
+            channel: Optional channel to deploy from.
+        """
+        opts = {}
+
+        if version is not None:
+            opts["gpu_operator_operatorhub_version"] = version
+            print(
+                f"Deploying the GPU Operator from OperatorHub using version '{version}'."
+            )
+
+        if channel is not None:
+            if version is None:
```

Comment:
just a quick "style" question, why `is None` / `is not None` and not `if channel` `if not version`?
this would mostly differ with `""` being allowed with your syntax and rejected by mine, but I see it as a bonus as version/channel shouldn't be an empty string anyway

Reply:
Because Fire is very opinionated:
![image](https://user-images.githubusercontent.com/10882062/126084006-cd27cc2f-bdbf-416b-b2c3-8ade21fb4607.png)

Reply:
(in other words, if one day you'll want to have a channel called `False` or `0` or `[]` then having `if channel` rather than `if channel is not None` will get you dropped.)

There's still an issue calling your channel `None` though...

Reply:
Also we'll have to stringify whatever parameter we get and not just pass it blindly in the dictionary for supporting a channel called `False` or `0`, but you get my point

Reply:
ack, thanks for the explanations!